### PR TITLE
[RAT] refactor chatter pressed

### PR DESCRIPTION
**Pull request by GrotheFAF**
lose the setDisabled() - insteat add only needed menu topics with menu_add()
factor out player/irc into f_action_connect()

(sadly there is no permanence in me.addF../remF..)

### CODE REVIEW

File: src/chat/chatter.py [hunk 1]
```diff
@@ -265,127 +265,79 @@ def doubleClicked(self, item):
...
-                elif url.scheme() == "faflive":
-                    actionReplay.setEnabled(True)
-
-        # Triggers
-        actionViewAliases.triggered.connect(self.viewAliases)
-        actionSelectAvatar.triggered.connect(self.selectAvatar)
-        actionReplay.triggered.connect(self.viewReplay)
-        actionVaultReplay.triggered.connect(self.viewVaultReplay)
-        actionJoin.triggered.connect(self.joinInGame)
+        def menu_add(action_str, action_connect, separator=None):
```
Comment (Wesmania) [inline src/chat/chatter.py]: If separator is a bool, we could make if default to False.

File: src/chat/chatter.py [hunk 2]
```diff
@@ -265,127 +265,79 @@ def doubleClicked(self, item):
...
-                actionCloseLobby = QtWidgets.QAction("Close FAF Client", menu)
-                menu.addAction(actionCloseLobby)
-                actionCloseLobby.triggered.connect(lambda: self.lobby.client.closeLobby(self.name))
+        # Replays in vault
+        if self.id != -1:  # not for irc user
+            menu_add("View Replays in Vault", self.viewVaultReplay, True)
 
-            menu.addSeparator()
+        # Friends and Foes Lists
+        def f_action_connect(f, irc_f):  # Irc or not Irc, that's the Question
```
Comment (Wesmania) [inline src/chat/chatter.py]: I don't know if f_action_connect is descriptive enough, how about player_or_irc_action?

**Review by Wesmania — COMMENTED** [review]
Looks really good, just a nitpick or two.

File: src/chat/chatter.py (hunk 2 above)
Comment (GrotheFAF) [reply src/chat/chatter.py]: it is the action_connector, got the f_ (Friends_/Foes_) to distinguish from the one in menu_add.

Comment (Wesmania) [reply src/chat/chatter.py]: I know what it does, I just feel that one can't tell what it does straight from its name. If we change it, we probably won't even need the comments explaining what it does. I think we even have some code guideline about such naming buried in client readme.

**Comment by Wesmania** [comment]
What do you mean by permanence? If it's some feature, we could go ahead and add an issue for it.

**Comment by GrotheFAF** [comment]
The user add/remFriend/Foe changes don't survive a Client restart.
(the clientwindow versions do, and the user add/remIrcFriend/Foe do)

File: src/chat/chatter.py (hunk 2 above)
Comment (GrotheFAF) [reply src/chat/chatter.py]: I like the comment.

Comment (GrotheFAF) [reply src/chat/chatter.py]: renamed it to player_or_irc_action_connect

File: src/chat/chatter.py (hunk 1 above)
Comment (GrotheFAF) [reply src/chat/chatter.py]: I had False at first, but None makes more sense.

Comment (GrotheFAF) [reply src/chat/chatter.py]: renamed it to False

**Pull request merged**